src/Update/PowerToys.Update.cpp

```diff
@@ -169,14 +169,34 @@ bool InstallNewVersionStage2(std::wstring installer_path, std::wstring_view inst
         }
     }
 
+    // Delete only .msi and .exe
+    // Skip the installer if installation failed
+    for (const auto& entry : fs::directory_iterator(updating::get_pending_updates_path()))
+    {
+        auto entryPath = entry.path().wstring();
+        std::transform(entryPath.begin(), entryPath.end(), entryPath.begin(), ::towlower);
+
+        if (entryPath.ends_with(L".msi") || entryPath.ends_with(L".exe"))
+        {
+            if (installer_path.find(entryPath) != std::string::npos && !success)
```

Comment:
```suggestion
// Skipping installer files of current update in case of update has failed.
            if (installer_path.find(entryPath) != std::string::npos && !success)
```